src/engraving/paint/debugpaint.cpp

```diff
@@ -80,3 +83,95 @@ void DebugPaint::paintElementsDebug(mu::draw::Painter& painter, const QList<Ms::
     debugger->restorePenColor();
     painter.setProvider(debugger->realProvider(), false);
 }
+
+void DebugPaint::paintPageDebug(Painter& painter, const Page* page)
+{
+    IF_ASSERT_FAILED(page) {
+        return;
+    }
+
+    Score* score = page->score();
+
+    IF_ASSERT_FAILED(score) {
+        return;
+    }
+
+    qreal scaling = painter.worldTransform().m11() / uiConfiguration()->guiScaling();
+
+    painter.save();
+
+    if (MScore::showSystemBoundingRect) {
+        painter.setBrush(BrushStyle::NoBrush);
+        painter.setPen(Pen(Color::black, 3.0 / scaling));
+
+        for (const System* system : page->systems()) {
+            PointF pt(system->ipos());
+            qreal h = system->height() + system->minBottom() + system->minTop();
+            painter.translate(pt);
+            RectF rect(0.0, -system->minTop(), system->width(), h);
+            painter.drawRect(rect);
+            painter.translate(-pt);
+        }
+    }
+
+    if (MScore::showSkylines) {
+        for (const System* system : page->systems()) {
+            for (SysStaff* ss : *system->staves()) {
+                if (!ss->show()) {
+                    continue;
+                }
+                PointF pt(system->ipos().x(), system->ipos().y() + ss->y());
+                painter.translate(pt);
+                ss->skyline().paint(painter, 3.0 / scaling);
+                painter.translate(-pt);
+            }
+        }
+    }
+
+    if (MScore::showSegmentShapes) {
+        painter.setBrush(BrushStyle::NoBrush);
+        painter.setPen(Pen(Color(238, 238, 144), 2.0 / scaling));
+
+        for (const System* system : page->systems()) {
+            for (const MeasureBase* mb : system->measures()) {
+                if (mb->type() == ElementType::MEASURE) {
+                    const Measure* m = static_cast<const Measure*>(mb);
+
+                    for (const Segment* s = m->first(); s; s = s->next()) {
+                        for (int i = 0; i < score->nstaves(); ++i) {
+                            PointF pt(s->pos().x() + m->pos().x() + system->pos().x(),
+                                      system->staffYpage(i));
+                            painter.translate(pt);
+                            s->shapes().at(i).paint(painter);
+                            painter.translate(-pt);
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+#ifndef NDEBUG
```

Comment:
The problem is that `Measure::corrupted()` and everything supporting it is hidden behind those `#ifndef NDEBUG` things. Should I remove them, so that those methods are also available in release builds?